File: src/DataScience/MATLAB/mvc.m
function result = mvc
%MVC Separate model state, controller input handling and view rendering.
model = struct("count", 0);
before = renderView(model);
model = controller(model, "increment");
after = renderView(model);
result = struct("before", before, "after", after, "count", model.count);
end

function model = controller(model, action)
if action == "increment"
    model.count = model.count + 1;
end
end

function text = renderView(model)
text = "count=" + string(model.count);
end

File: src/Web/JavaScriptJS/patterns/mvc.js
const assert = require("node:assert/strict");
/** Separates model state, view rendering and controller input handling. */
function run() {
  const model = { count: 0 };
  const view = { render: (m) => `count=${m.count}` };
  const controller = { increment() { model.count += 1; return view.render(model); } };
  assert.equal(controller.increment(), "count=1");
  return "mvc";
}
module.exports = { run };
if (require.main === module) run();
